src/mudlet.cpp

```diff
@@ -2341,16 +2352,21 @@ void mudlet::slot_notes()
 
 void mudlet::slot_irc()
 {
-    if( ! mpIRC )
-    {
-        mpIRC = new dlgIRC();
-        mpIRC->setWindowTitle( tr( "Mudlet live IRC Help Channel #mudlet-help on irc.freenode.net" ) );
-        mpIRC->setWindowIcon( QIcon( QStringLiteral( ":/icons/mudlet_irc.png" ) ) );
-        mpIRC->resize(660,380);
+    Host * pHost = getActiveHost();
+    bool isDefaultHost = false;
+    if( ! pHost ) {
+        pHost = mpDefaultHost;
+        isDefaultHost = true;
     }
 
-    mpIRC->raise();
-    mpIRC->show();
+    if( ! mpIrcClientMap.contains(pHost) )
+    {
+        QPointer<dlgIRC> dlg = new dlgIRC(pHost);
+        dlg->setDefaultHostClient(isDefaultHost);
+        mpIrcClientMap[pHost] = dlg;
```

Comment:
🤔  I am not SURE but I have a suspicion that you might need:
 
    mpIrcClientMap[pHost] = dlg.data();

here - I cannot remember the exact details why - which is frustrating - but I has a crash in this area when I tried to to fire up a second IRC session for a second profile... but didn't when I added the `.data()` however one run is not conclusive proof... and I'd welcome others testing they can start a separate IRC session on separate `Host` instances (i.e. in a multi-playing situation).